controllers/code_lookup.go

```diff
@@ -0,0 +1,50 @@
+package controllers
+
+import (
+	"encoding/json"
+	"github.com/intervention-engine/fhir/server"
+	"github.com/intervention-engine/ie/utilities"
+	"gopkg.in/mgo.v2/bson"
+	"net/http"
+)
+
+type code_request_form struct {
+	CodeSystem  string `json:"codesystem"`
+	Query       string `json:"query"`
+	ResultLimit int    `json:"limit"`
+}
+
+func CodeLookup(rw http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+
+	var codereqform code_request_form
+
+	err := decoder.Decode(&codereqform)
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusBadRequest)
+	}
+
+	codesystem, querystring := codereqform.CodeSystem, codereqform.Query
+	resultlimit := codereqform.ResultLimit
```

Comment:
This is a little nit-picky, but I think this would be simplified if you just called the `code_request_form` variable `f` instead of `codereqform`, then _skipped_ creating these intermediate variables here and just referenced `f.CodeSystem`, `f.Query`, and `f.ResultLimit` in the query.